eng/Subsets.props

```diff
@@ -124,7 +124,8 @@
     <_NativeAotSupportedOS Condition="'$(TargetOS)' == 'windows' or '$(TargetOS)' == 'linux' or '$(TargetOS)' == 'osx' or '$(TargetOS)' == 'maccatalyst' or '$(TargetOS)' == 'iossimulator' or '$(TargetOS)' == 'ios' or '$(TargetOS)' == 'tvossimulator' or '$(TargetOS)' == 'tvos' or '$(TargetOS)' == 'freebsd'">true</_NativeAotSupportedOS>
     <_NativeAotSupportedArch Condition="'$(TargetArchitecture)' == 'x64' or '$(TargetArchitecture)' == 'arm64' or '$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'loongarch64' or ('$(TargetOS)' == 'windows' and '$(TargetArchitecture)' == 'x86')">true</_NativeAotSupportedArch>
     <NativeAotSupported Condition="'$(_NativeAotSupportedOS)' == 'true' and '$(_NativeAotSupportedArch)' == 'true'">true</NativeAotSupported>
-    <UseNativeAotForComponents Condition="'$(NativeAotSupported)' == 'true' and '$(TargetOS)' == '$(HostOS)' and '$(TargetsLinuxBionic)' != 'true'">true</UseNativeAotForComponents>
+    <_IsCommunityArchitecture Condition="'$(CrossBuild)' == 'true' and ('$(TargetArchitecture)' == 'loongarch64' or '$(TargetArchitecture)' == 'riscv64')">true</_IsCommunityArchitecture>
```

Comment:
I would expect the community architectures to be able to `UseNativeAotForComponents` here if they have otherwise working NAOT support.

What was wrong with the original condition?

Reply:
This is in line with what we have been using thus far.

1. linux-loongarch64 folks are not using -cross so they have not observed this issue https://github.com/dotnet/runtime/issues/108973#issuecomment-2541077363
2. freebsd-x64 is protected by TargetOS == HostOS condition below

When building on the device (CrossBuild=false) with previous SDK version from global.json, there is no problem and we don't need multi-stages. Otherwise, we will need to add stage3 as stage2 is the first one publishes ILCompiler.csproj.

Reply:
We only need an ILCompiler that runs on host. What prevents it from being produced in the first stage?

Reply:
Stage 1 does not publish ILCompiler and crossgen2 because they need live apphost build. Stage 2 publishes them as single file using live apphost from stage 1.